### PR TITLE
[wasm] Add a `--wasmDataDir` parameter

### PR DESCRIPTION
**Pull request by radical**
In an earlier commit, generated projects got support for importing
props/targets files from the benchmark projects. And that allows
customizing the build.

But there isn't a clean way to inject/add references to the build of the
project, like custom paths, or access to additional files, without
adding new command line parameters.

To solve that, `--wasmDataDir` param is added which is surfaced in the
generated project as the msbuild property - `$(WasmDataDir)`.

An example use case is to have the wasm js file added to this data dir,
and in the props file use:
`<WasmMainJSPath>$(WasmDataDir)test-main.js</WasmMainJSPath>`.

### CODE REVIEW

**Comment by radical** [comment]
cc @adamsitnik @lewing @radekdoulik

**Review by radekdoulik — APPROVED** [review]

File: src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs:191]: are there any scenarios in which this argument is mandatory? If so the following method should take that into account:

https://github.com/dotnet/BenchmarkDotNet/blob/5852a1340653250f1fa8801102556026d8f87e45/src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs#L100

File: src/BenchmarkDotNet/Environments/Runtimes/WasmRuntime.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Environments/Runtimes/WasmRuntime.cs:29]: thank you for adding xml doc comment 👍 (most of our contributors ignore build warnings and don't fix them)

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @radical !

Before I hit the merge button could you please respond to my question?

File: src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs
Comment (radical) [reply src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs:191]: It's not mandatory.

**Pull request merged**
